packages/cice5/package.py

```diff
@@ -0,0 +1,218 @@
+
+    # The integr represents environment variable NTASK
+    __targets = {24: {}, 480: {}, 722: {}, 1682: {}}
+    __targets[24]["driver"] = "auscom"
+    __targets[24]["grid"] = "360x300"
+    __targets[24]["blocks"] = "24x1"
+
+    __targets[480]["driver"] = "auscom"
+    __targets[480]["grid"] = "1440x1080"
+    __targets[480]["blocks"] = "48x40"
+
+    # Comment from bld/config.nci.auscom.3600x2700:
+    # Recommendations:
+    #   use processor_shape = slenderX1 or slenderX2 in ice_in
+    #   one per processor with distribution_type='cartesian' or
+    #   squarish blocks with distribution_type='rake'
+    # If BLCKX (BLCKY) does not divide NXGLOB (NYGLOB) evenly, padding
+    # will be used on the right (top) of the grid.
+    __targets[722]["driver"] = "auscom"
+    __targets[722]["grid"] = "3600x2700"
+    __targets[722]["blocks"] = "90x90"
+
+    __targets[1682]["driver"] = "auscom"
+    __targets[1682]["grid"] = "3600x2700"
+    __targets[1682]["blocks"] = "200x180"
+
+    def edit(self, spec, prefix):
+
+        srcdir = self.stage.source_path
+        makeinc_path = join_path(srcdir, "bld", "Macros.spack")
+
+        config = {}
+
+        incs = (spec["oasis3-mct"].headers).cpp_flags + "/psmile.MPI1" + " "
+        for l in ["parallelio", "oasis3-mct", "libaccessom2", "netcdf-fortran"]:
+            incs += (spec[l].headers).cpp_flags + " "
+
+        # NOTE: The order of the libraries matter during the linking step!
+        # NOTE: datetime-fortran is a dependency of libaccessom2.
+        libs = "-L" + (spec["parallelio"].prefix) + "/lib -lpiof -lpioc "
+        for l in ["oasis3-mct", "libaccessom2", "netcdf-c", "netcdf-fortran", "datetime-fortran"]:
+            libs += (spec[l].libs).ld_flags + " "
+
+### Specialty code
+# setenv DITTO    no        # reproducible diagnostics
+# setenv THRD     no        # set to yes for OpenMP threading
+# if ( $THRD == 'yes') setenv OMP_NUM_THREADS 2 # positive integer 
+# if ($driver == 'access') then
+#     setenv ACCESS   yes
+# else
+#     setenv ACCESS   no
+# endif
+# setenv NICELYR    4       # number of vertical layers in the ice
+# setenv NSNWLYR    1       # number of vertical layers in the snow
+# setenv NICECAT    5       # number of ice thickness categories
+
+        # Copied from bld/Macros.nci
+        config["pre"] = f"""
+INCLDIR    := -I. {incs}
+SLIBS      := {libs}
+ULIBS      :=
+CPP        := cpp
+FC         := mpifort
+
+CPPFLAGS   := -P -traditional
+CPPDEFS    := -DLINUX -DPAROPT
+CFLAGS     := -c -O2
+FIXEDFLAGS := -132
+FREEFLAGS  :=
+"""
+
+        config["gcc"] = f"""
+# TODO: removed -std=f2008 due to compiler errors
+FFLAGS = -Wall -fdefault-real-8 -fdefault-double-8 -ffpe-trap=invalid,zero,overflow -fallow-argument-mismatch
+"""
+
+        # module load intel-compiler/2019.5.281
+        config["intel"] = f"""
+NCI_INTEL_FLAGS := -r8 -i4 -traceback -w -fpe0 -ftz -convert big_endian -assume byterecl -check noarg_temp_created
```

Comment:
This will not be reproducible if `-traceback` is always on right? 

Do you want to make a `nodebuginfo` variant as you did with `mom5`?

Reply:
1) No, -traceback and -debug all result in a non-deterministic build.
2) I assume you are referring to oasis3-mct and not mom5. I hoping that https://github.com/ACCESS-NRI/ACCESS-OM/issues/12 can be finalized first. I'm planning on using the variant name "deterministic" for builds that will be used to do a binary comparison.